protocols/mdns/src/behaviour.rs

```diff
@@ -118,27 +122,52 @@ pub struct Mdns {
 
     /// Discovery timer.
     timeout: Timer,
+
+    // Mutlicast address.
+    multicast_addr: IpAddr,
 }
 
 impl Mdns {
     /// Builds a new `Mdns` behaviour.
     pub async fn new(config: MdnsConfig) -> io::Result<Self> {
-        let recv_socket = {
-            let socket = Socket::new(Domain::IPV4, Type::DGRAM, Some(socket2::Protocol::UDP))?;
-            socket.set_reuse_address(true)?;
-            #[cfg(unix)]
-            socket.set_reuse_port(true)?;
-            socket.bind(&SocketAddr::new(Ipv4Addr::UNSPECIFIED.into(), 5353).into())?;
-            let socket = UdpSocket::from(socket);
-            socket.set_multicast_loop_v4(true)?;
-            socket.set_multicast_ttl_v4(255)?;
-            Async::new(socket)?
+        let recv_socket = match config.multicast_addr {
+            IpAddr::V4(_) => {
+                let socket = Socket::new(Domain::IPV4, Type::DGRAM, Some(socket2::Protocol::UDP))?;
+                socket.set_reuse_address(true)?;
+                #[cfg(unix)]
+                socket.set_reuse_port(true)?;
```

Comment:
If two processes using this approach run on the same host, either of them sees only a stochastic selection of half of the received datagrams — isn’t this going to break some assumptions?

Reply:
I think you're forgetting that it's multicast. So all sockets get all messages [0]

- [0] https://stackoverflow.com/questions/18443004/usage-of-so-reuseport-with-multicast-udp

Reply:
While what you say sounds plausible, I have been unable to find any normative reference that describes the interplay between IP_ADD_MEMBERSHIP and SO_REUSEPORT. The SO link you posted also contains the same question, unanswered.

Reply:
I guess experimental validation of a hypothisis is sufficient (so we don't have to dive into the kernel sources)?

Running two instances of `export RUST_LOG=libp2p_mdns cargo run --example mdns-passive-discovery` on the same machine should convince you that both receive the request.